src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.mongodb.client.MongoCursor;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.bson.BsonDocument;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MongoResultSetTests {
+    private static final List<String> FIELDS = List.of("id", "title", "author");
+
+    @Mock
+    private MongoCursor<BsonDocument> mongoCursor;
+
+    private MongoResultSet mongoResultSet;
+
+    @BeforeEach
+    void setUp() {
+        mongoResultSet = new MongoResultSet(mongoCursor, FIELDS);
+    }
+
+    @Nested
+    class CloseTests {
+
+        @BeforeEach
+        void setUp() throws SQLException {
+            mongoResultSet.close();
+        }
+
+        @Test
+        @DisplayName("No-op when 'close()' is called on a closed MongoResultSet")
+        void testNoopWhenCloseStatementClosed() throws SQLException {
+            // when && then
+            assertDoesNotThrow(() -> mongoResultSet.close());
+        }
+
+        @ParameterizedTest(name = "SQLException is thrown when \"{0}\" is called on a closed MongoResultSet")
+        @MethodSource("getMongoResultSetMethodInvocationsImpactedByClosing")
+        void testCheckClosed(String label, ResultSetMethodInvocation methodInvocation) throws SQLException {
+            // when && then
```

Comment:
This comment is confusing. I am familiar with the given/when/then pattern, however here there is no given. I think we should avoid writing these, except in cases where they help to separate the parts of a test.

Reply:
the `@BeforeEach` below serves as `given`, but I agree, we don't need such delimiters when they are together. Removed.
```
@BeforeEach
void setUp() throws SQLException {
    mongoResultSet.close();
}
```